Template/Trigger/ElementVisibilityTrigger.web.js

```diff
@@ -235,9 +230,82 @@
             };
         }
 
+        function getSelectors() {
+            var selectionMethod = parameters.get('selectionMethod');
+            if (selectionMethod === 'elementId') {
+                return '#' + parameters.get('elementId');
+            } else if (selectionMethod === 'cssSelector') {
+                return parameters.get('cssSelector');
+            }
+
+            return;
+        }
+
+        function setIntersectionObserver(triggerEvent) {
+            if ('IntersectionObserver' in window) {
+                var interSectionObserverOptions = {
+                    root: null, // document's viewport as the container.
+                    rootMargin: '0px',
+                    threshold: (minPercentVisible / 100)
+                };
+                var observerIntersection = new IntersectionObserver(function (entries) {
+                    interSectionCallback(entries, triggerEvent, observerIntersection);
+                }, interSectionObserverOptions);
+
+                if (selectors) {
+                    TagManager.dom.bySelector(selectors).forEach(function (element) {
+                        observerIntersection.observe(element);
+                    });
+                }
+            }
+
+        }
+
+        function interSectionCallback(entries, triggerEvent, observer) {
+            var dom = TagManager.dom;
+            entries.forEach(function (entry) {
+                if (entry.intersectionRatio > 0) {
+                    if (blockTrigger || (onlyOncePerElement && isNodeEventTriggered(entry.target))) {
+                        return;
+                    }
+                    triggerEvent({
+                        event: 'mtm.ElementVisibility',
+                        'mtm.elementVisibilityPercentage': minPercentVisible,
```

Comment:
Is there a way to get the actual percentage with intersection observer? is it maybe `intersectionRatio`? Be great to get the actual value there

Reply:
otherwise we could fallback to `getPercentVisible(node)` maybe

Reply:
Started using getPercentVisible along with max of minPercentVisible as discussed